services/s3/src/test/java/software/amazon/awssdk/services/s3/internal/crossregion/S3CrossRegionSyncClientTest.java

```diff
@@ -67,7 +68,7 @@ public void before() {
     public void standardOp_crossRegionClient_noOverrideConfig_SuccessfullyIntercepts() {
         S3Client crossRegionClient = new S3CrossRegionSyncClient(s3Client);
         crossRegionClient.getObject(r -> r.bucket(BUCKET).key(KEY));
-        assertThat(captureInterceptor.endpointProvider).isInstanceOf(S3CrossRegionSyncClient.BucketEndpointProvider.class);
+        assertThat(captureInterceptor.endpointProvider).isInstanceOf(BucketEndpointProvider.class);
     }
```

Comment:
Can we add a couple of tests in this test class too that verifies redirects since this class is more of a functional test that doesn't mock the client?

Reply:
Will continue adding , added some parameterized test cases